store/src/cell.rs

```diff
@@ -26,6 +31,39 @@ use std::collections::HashMap;
 pub fn attach_block_cell(txn: &StoreTransaction, block: &BlockView) -> Result<(), Error> {
     let transactions = block.transactions();
 
+    // update cells root mmr
+    let block_number = block.header().number();
+    let mut cells_root_mmr = txn.cells_root_mmr(block_number);
+    for tx in transactions.iter() {
+        for input in tx.inputs().into_iter() {
+            let out_point = input.previous_output();
+            // cellbase and genesis block's tx may not have previous output
+            if let Some(mut cell_status) = txn.get_cells_root_mmr_status(&out_point) {
+                cells_root_mmr
+                    .update(
+                        cell_status.mmr_position,
+                        hash_out_point_and_status(&out_point, cell_status.created_by, block_number),
+                    )
+                    .map_err(|e| InternalErrorKind::MMR.other(e))?;
+                cell_status.mark_as_consumed(block_number);
+                txn.insert_cells_root_mmr_status(&out_point, &cell_status)?;
+            }
+        }
+
+        for out_point in tx.output_pts().into_iter() {
+            let hash = hash_out_point_and_status(&out_point, block_number, BlockNumber::MAX);
+            let mmr_position = cells_root_mmr
+                .push(hash)
+                .map_err(|e| InternalErrorKind::MMR.other(e))?;
+            let cell_status = CellStatus::new(mmr_position, block_number);
+            txn.insert_cells_root_mmr_status(&out_point, &cell_status)?;
```

Comment:
Ensure to add test cases that cells created in the block are consumed in other transactions in the same block.